segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/ChunkedSegmentStorage.java

```diff
@@ -653,6 +669,58 @@ private void close(String message, AutoCloseable toClose) {
         }
     }
 
+    /**
+     * Adds block index entries.
+     */
+    void addBlockIndexEntries(MetadataTransaction txn, String segmentName, long startOffset, long endOffset, List<ChunkNameOffsetPair> readIndexEntries) {
+        val blockSize = config.getIndexBlockSize();
+        val startBlock = startOffset / blockSize;
+        val endBlock = endOffset / blockSize;
+        TreeMap<Long, String> map = new TreeMap<>();
+        readIndexEntries.stream().forEach( pair -> map.put(pair.getOffset(), pair.getChunkName()));
```

Comment:
Can we do this more efficiently by copying the entries in a new list, sorting it, then parsing it in order. 

From what I see, you loop through each entry, then do a `.floor` in each loop iteration. Would the suggested change improve it?

Reply:
Fixed.